HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - The recording was found but it could not be written to the
+    response. Or there was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v1/targets/localhost/recordings/foo --output foo.jfr
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  530k    0  530k    0     0  9303k      0 --:--:-- --:--:-- --:--:-- 9303k
+    ```
+
+* #### `TargetRecordingPatchHandler`
+
+    ###### synopsis
+    **STOP**
+
+    Stops a recording in a target JVM.
+
+    **SAVE**
+
+    Saves a recording in a target JVM to persistent storage.
+    The default directory used is `/flghtrecordings`, but the environment
+    variable `CONTAINER_JFR_ARCHIVE_PATH` can be used to specify a different
+    path.
+
+    ###### request
+    `PATCH /api/v1/targets/:targetId/recordings/:recordingName`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+
+    `recordingName` - The name of the recording to patch.
+
+    The body must be either `STOP`, to stop the recording,
+    or `SAVE`, to save the recording (case insensitive).
+
+    ###### response
+    **General**
+
+    `400` - The operation is unsupported. The body is an error message.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `404` - The target or the recording could not be found.
+    The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    **STOP**
+
+    `200` - No body.
+
+    `500` - The recording could not be stopped. The body is an error message.
+
+    **SAVE**
+
+    `200` - The body is the name of the recording that was saved.
+    Note that this name will be different from the recording's original name,
+    to add metadata.
+
+    ###### example
+    ```
+    $ curl -X PATCH --data "STOP" localhost:8181/api/v1/targets/localhost/recordings/foo
+
+    $ curl -X PATCH --data "SAVE" http://localhost:8181/api/v1/targets/localhost/recordings/foo
+    localhost_foo_20200911T155146Z.jfr
+    ```
+
+
+* #### `TargetRecordingsGetHandler`
+
+    ###### synopsis
+    Returns a list of all the recordings in a target JVM.
+
+    ###### request
+    `GET /api/v1/targets/:targetId/recordings`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+
+    ###### response
+    `200` - The body consists of a comma-seperated list of recordings,
+    enclosed in square brackets.
+
+    The format for a recording is
+    `{"downloadUrl":"$DOWNLOAD_URL","name":"$NAME","reportUrl":"$REPORT_URL"}`.
```

Comment:
I think this should include all of the attributes of a recording, so including `state`, `startTime`, `duration`, etc., no?

Reply:
Yeah I think I just copied it from the non-targeted version not realizing it's different.